great_expectations/core/batch.py

```diff
@@ -595,6 +595,15 @@ def head(self, n_rows=5, fetch_all=False):
         return self._data.execution_engine.resolve_metrics((metric,))[metric.id]
 
 
+def get_batch_request_as_dict(
+    batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None
+) -> Optional[dict]:
+    if isinstance(batch_request, (BatchRequest, RuntimeBatchRequest)):
+        batch_request = batch_request.to_dict()
+
+    return batch_request
```

Comment:
@cdkini I put a comment explaining this in the main schema class:
```
class RuleBasedProfilerConfigSchema(Schema):
    """
    Schema classes for configurations which extend from BaseYamlConfig must extend top-level Marshmallow Schema class.
    Schema classes for their constituent configurations which extend DictDot leve must extend NotNullSchema class.
    """
```
Happy to jump on a call to give more background.  Thanks.

Reply:
@cdkini Secondly, the type annotations to become `Config` -- would require a separate redesign as this is a deviation from the current design found in DataContext and Checkpoint structures.

Reply:
> In the case batch_request is None, this will cause an error (cannot isinstance None).
> 
> ```python
> if batch_request is not None and and isinstance(...)
> ```

@cdkini The check 
```
    if isinstance(batch_request, (BatchRequest, RuntimeBatchRequest)):
```
Will return `False` on `None` -- do you agree?  I can strengthen it.!